### ISSUE TITLE
How to calculate the ray loss LM

### ISSUE THREAD

**Issue by zhangqianqianhzlh** [ByteDance-Seed/Depth-Anything-3#195]
The paper didn't mention how LM is calculated. Can you provide more info?

**Comment by thiesk** [comment]
While it's not explicitly written out exactly how L_M is calculated, the rays are optimized in both the L_M and the L_P loss. 

In L_M, it is the pixel-wise L_1 norm of the predicted ray map R^ and the ground truth M:
$$\mathcal{L}_M = \|\hat{R} - M\|_1$$

The GT ray map M is calculated from the known camera intrinsics and extrinsics from the training data (this is described in section 3.1 "Depth-ray representation"). Specifically:
* For each pixel $p$, the ray consists of the origin $t$ and direction $d = RK^{-1}p$.
* $d$ is kept unnormalized to preserve the scale.
* In the L_P loss, both the direction $d$ and origin $t$ are further optimized alongside depth to ensure the final point map $P = t + D \cdot d$ is correct. 

Before the loss is calculated, the GT signals are also normalized by a global scale factor (the mean L_2 norm of the point maps) to keep training stable.
